sources/dicom/large_image_source_dicom/assetstore/dicomweb_assetstore_adapter.py

```diff
@@ -0,0 +1,220 @@
+        doc['readOnly'] = True
+
+        required_fields = [
+            'url',
+        ]
+
+        info = doc.get(DICOMWEB_META_KEY, {})
+
+        for field in required_fields:
+            if field not in info:
+                raise ValidationException('Missing field: ' + field)
+
+        # If these are empty, they need to be converted to None
+        convert_empty_fields_to_none = [
+            'qido_prefix',
+            'wado_prefix',
+            'auth_type',
+        ]
+
+        for field in convert_empty_fields_to_none:
+            if isinstance(info.get(field), str) and not info[field].strip():
+                info[field] = None
+
+        # Now, if there is no authentication, verify that we can connect to the server.
+        # If there is authentication, we may need to prompt the user for their
+        # username and password sometime before here.
+        if info['auth_type'] is None:
+            study_instance_uid_tag = '0020000D'
+            series_instance_uid_tag = '0020000E'
+
+            client = _create_dicomweb_client(info)
+            # Try to search for series. If we get an http error, raise
+            # a validation exception.
+            try:
+                series = client.search_for_series(
+                    limit=1,
+                    fields=(study_instance_uid_tag, series_instance_uid_tag),
+                )
+            except HTTPError as e:
+                raise ValidationException('Failed to validate DICOMweb server settings: ' + str(e))
+
+            # If we found a series, then test the wado prefix as well
+            if series:
+                # The previous query should have obtained uids for a specific
+                # study and series.
+                study_uid = series[0][study_instance_uid_tag]['Value'][0]
+                series_uid = series[0][series_instance_uid_tag]['Value'][0]
+                try:
+                    # Retrieve the metadata of this series as a wado prefix test
+                    client.retrieve_series_metadata(
+                        study_instance_uid=study_uid,
+                        series_instance_uid=series_uid,
+                    )
+                except HTTPError as e:
+                    raise ValidationException('Failed to validate DICOMweb WADO prefix: ' + str(e))
+
+        return doc
+
+    def initUpload(self, upload):
+        msg = 'DICOMweb assetstores are import only.'
+        raise NotImplementedError(msg)
+
+    def finalizeUpload(self, upload, file):
+        msg = 'DICOMweb assetstores are import only.'
+        raise NotImplementedError(msg)
+
+    def deleteFile(self, file):
+        # We don't actually need to do anything special
+        pass
+
+    def downloadFile(self, file, offset=0, headers=True, endByte=None,
+                     contentDisposition=None, extraParameters=None, **kwargs):
+        # FIXME: do we want to support downloading files? We probably
```

Comment:
We eventually _do_ want to support this (in some future PR).  When we pass files to workers, if this supports offset, then there are some transfer efficiencies if we don't need a complete file.

Trying to run a HistomicsTK task (like nuclei detection) on one of the imported files will tell us what we need to do for basic functionality and that will determine how valuable this is.

Reply:
This is now tracked in #1309.